modules/shared/src/main/scala/org/tessellation/schema/hex.scala

```diff
@@ -0,0 +1,103 @@
+package org.tessellation.schema
+
+import java.nio.charset.StandardCharsets
+import java.security.spec.X509EncodedKeySpec
+import java.security.{KeyFactory, PublicKey}
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.key.{ECDSA, PublicKeyHexPrefix}
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.boolean.And
+import eu.timepit.refined.collection.Size
+import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.Even
+import eu.timepit.refined.refineV
+import eu.timepit.refined.scalacheck.arbitraryRefType
+import eu.timepit.refined.string.MatchesRegex
+import org.scalacheck.{Arbitrary, Gen}
+import shapeless.Witness
+
+object hex {
+
+  type HexStringSpec = And[MatchesRegex["^[0-9a-f]*$"], Size[Even]]
+  type HexString = Refined[String, HexStringSpec]
```

Comment:
Sometimes you have `Refined[String, A]` and sometimes `String Refined A`.
I think we should unify it. My personal preference is ltr but up to you:
```
String Refined A
A And B
```

instead of

```
Refined[String, A]
And[A, B]
```

Reply:
True, I will switch to infix-only notation.